spec/cop/sorbet/signatures/enforce_signatures_spec.rb

```diff
@@ -0,0 +1,227 @@
+        sig { void }
+        def foo; end
+      RUBY
+    end
+
+    it 'makes offense if a top-level method has no signature' do
+      expect_offense(<<~RUBY)
+        def foo; end
+        ^^^^^^^^^^^^ Each method is required to have a signature.
+      RUBY
+    end
+
+    it 'does not check signature validity' do # Validity will be checked by Sorbet
+      expect_no_offenses(<<~RUBY)
+        sig { foo(bar).baz }
+        def foo; end
+      RUBY
+    end
+
+    it 'makes no offense if a method has a signature' do
+      expect_no_offenses(<<~RUBY)
+        class Foo
+          sig { void }
+          def foo1; end
+        end
+      RUBY
+    end
+
+    it 'makes offense if a method has no signature' do
+      expect_offense(<<~RUBY)
+        class Foo
+          def foo; end
+          ^^^^^^^^^^^^ Each method is required to have a signature.
+        end
+      RUBY
+    end
+
+    it 'makes no offense if a singleton method has a signature' do
+      expect_no_offenses(<<~RUBY)
+        class Foo
+          sig { void }
+          def self.foo1; end
+        end
+      RUBY
+    end
+
+    it 'makes offense if a singleton method has no signature' do
+      expect_offense(<<~RUBY)
+        class Foo
+          def self.foo; end
+          ^^^^^^^^^^^^^^^^^ Each method is required to have a signature.
+        end
+      RUBY
+    end
+
+    it 'makes no offense if an accessor has a signature' do
+      expect_no_offenses(<<~RUBY)
+        class Foo
+          sig { returns(String) }
+          attr_reader :foo
+          sig { params(bar: String).void }
+          attr_writer :bar
+          sig { params(bar: String).returns(String) }
+          attr_accessor :baz
+        end
+      RUBY
+    end
+
+    it 'makes offense if an accessor has no signature' do
+      expect_offense(<<~RUBY)
+        class Foo
+          attr_reader :foo
+          ^^^^^^^^^^^^^^^^ Each method is required to have a signature.
+          attr_writer :bar
+          ^^^^^^^^^^^^^^^^ Each method is required to have a signature.
+          attr_accessor :baz
+          ^^^^^^^^^^^^^^^^^^ Each method is required to have a signature.
+        end
+      RUBY
+    end
+
+    it 'does not check the signature for accessors' do # Validity will be checked by Sorbet
+      expect_no_offenses(<<~RUBY)
+        class Foo
+          sig { void }
+          attr_reader :foo, :bar
+        end
+      RUBY
+    end
+
+    shared_examples_for('autocorrect with config') do
+      it('autocorrects methods by adding signature stubs') do
+        expect(
+          autocorrect_source(<<~RUBY)
+            def foo; end
+            def bar(a, b = 2, c: Foo.new); end
+            def baz(&blk); end
+            def self.foo(a, b, &c); end
+          RUBY
+        ).to(eq(<<~RUBY))
+          sig { returns(T.untyped) }
+          def foo; end
+          sig { params(a: T.untyped, b: T.untyped, c: T.untyped).returns(T.untyped) }
+          def bar(a, b = 2, c: Foo.new); end
```

Comment:
The full list of different types of arguments is like this (from `tapioca` code):

```ruby
            when :req
              name.to_s
            when :opt
              "#{name} = _"
            when :rest
              "*#{name}"
            when :keyreq
              "#{name}:"
            when :key
              "#{name}: _"
            when :keyrest
              "**#{name}"
            when :block
              "&#{name}"
            end
```
We seem to be missing cases of splats (`*d`), kwargs (`**e`) and required keyword arguments (`f:`). Can we add them too please?

Reply:
Not sure what you mean as I donc think you can express those in Sorbet syntax? See https://sorbet.org/docs/error-reference#7019 for example.

I added a test to show the behaviour of this cop nonetheless: 
https://github.com/Shopify/rubocop-sorbet/pull/13/files#diff-d7dfba5ec2240af1506fc3c02f5e3d69R123-R126

Reply:
Yes, that added test was exactly what I meant. Thanks.